// @mondaycom/apps-cli/dist/consts/urls.js
const BASE_APPS_URL = '/api/apps';
export const appVersionIdBaseUrl = (appVersionId) => {
    return `/api/code/${appVersionId}`;
};
export const getAppVersionDeploymentStatusUrl = (appVersionId) => {
    return `${appVersionIdBaseUrl(appVersionId)}/deployments`;
};
export const getDeploymentSignedUrl = (appVersionId) => {
    return `${appVersionIdBaseUrl(appVersionId)}/deployments/signed-url`;
};
export const getLogsStreamForAppVersionIdUrl = (appVersionId, logsType, logsFilterCriteria) => {
    const logsFilterCriteriaParams = logsFilterCriteria
        ? `&fromDate=${logsFilterCriteria.fromDate.getTime()}&toDate=${logsFilterCriteria.toDate.getTime()}&text=${encodeURIComponent(logsFilterCriteria.text || '')}`
        : '';
    return `${appVersionIdBaseUrl(appVersionId)}/logs?type=${logsType}${logsFilterCriteriaParams}`;
};
export const listAppsUrl = () => {
    return BASE_APPS_URL;
};
export const listAppVersionsByAppIdUrl = (appId) => {
    return `${BASE_APPS_URL}/${appId}/versions`;
};
export const appEnvironmentUrl = (appId, key) => {
    return `/api/code/${appId}/env/${key}`;
};
export const appEnvironmentKeysUrl = (appId) => {
    return `/api/code/${appId}/env-keys`;
};
export const appReleasesUrl = (appVersionId) => {
    return `/apps_ms/app-versions/${appVersionId}/releases`;
};
